src/Bundle/DependencyInjection/Compiler/WinzouStateMachinePass.php

```diff
@@ -39,9 +38,12 @@ public function process(ContainerBuilder $container): void
         $container->setAlias('sm.callback.cascade_transition', CascadeTransitionCallback::class);
```

Comment:
IMHO this should be changed (tested the current proposal & it didn't work for me).

```suggestion
        $container->setAlias('sm.factory', FactoryInterface::class)->setPublic(true);
        $container->setAlias('sm.callback_factory', CallbackFactoryInterface::class)->setPublic(true);
        $container->setAlias('sm.callback.cascade_transition', CascadeTransitionCallback::class)->setPublic(true);
```

Reply:
PR seems fine as is for me, this is coming out of my local Sylius clone with my WIP upgrade PR applied:

```sh
Michaels-MacBook-Pro-2:Sylius mbabker$ bin/console debug:container sm.factory

 // This service is a public alias for the service SM\Factory\Factory
```

Compared to the current state of affairs:

```sh
Michaels-MacBook-Pro-2:Sylius mbabker$ bin/console debug:container sm.factory

 // This service is a private alias for the service SM\Factory\Factory                                                  

... snip ...

 ! [NOTE] The "sm.factory" service or alias has been removed or inlined when the container was compiled.
```